src/expression/ekat_expression_binary_op.hpp

```diff
@@ -52,24 +55,27 @@ class BinaryExpression : public Expression<BinaryExpression<ELeft,ERight,OP>>{
     }
   }
 
-  static auto ret_type () { return ELeft::ret_type() + ERight::ret_type(); }
+  static auto ret_type () {
+    return std::common_type_t<ret_left,ret_right>(0);
+  }
```

Comment:
I'm still confused about this `ret_type` function. Is it just returning a default element? Why not just define a public type 
```
using ret_type = std::common_type_t<ret_left,ret_right>;
```
Maybe I'm missing where it's being used..

Reply:
Sorry, I thought it deleted my comment. That was from looking at one of the first commits lol